pkg/trace/api/telemetry_test.go

```diff
@@ -0,0 +1,158 @@
+package api
+
+import (
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"regexp"
+	"strings"
+	"sync/atomic"
+	"testing"
+
+	traceconfig "github.com/DataDog/datadog-agent/pkg/trace/config"
+	"github.com/stretchr/testify/assert"
+)
+
+func assertingServer(t *testing.T, onReq func(req *http.Request, reqBody []byte) error) *httptest.Server {
+	return httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		body, err := ioutil.ReadAll(req.Body)
+		assert.NoError(t, err)
+		assert.NoError(t, onReq(req, body))
+		_, err = w.Write([]byte("OK"))
+		assert.NoError(t, err)
+		req.Body.Close()
+	}))
+}
+
+func TestTelemetryBasicProxyRequest(t *testing.T) {
+	srv := assertingServer(t, func(req *http.Request, body []byte) error {
+		assert.Equal(t, "body", string(body), "invalid request body")
+		assert.Equal(t, "test_apikey", req.Header.Get("DD-API-KEY"))
+		assert.Equal(t, "test_hostname", req.Header.Get("DD-Agent-Hostname"))
+		assert.Equal(t, "test_env", req.Header.Get("DD-Agent-Env"))
+		assert.Equal(t, "/path", req.URL.Path)
+
+		return nil
+	})
+	defer mockConfigMap(map[string]interface{}{
+		"api_key":                     "test_apikey",
+		"apm_config.telemetry.dd_url": srv.URL,
+		"hostname":                    "test_hostname",
+		"skip_ssl_validation":         true,
+		"env":                         "test_env",
+	})() // reset config after the test
+
+	req, err := http.NewRequest("POST", "/telemetry/proxy/path", strings.NewReader("body"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	rec := httptest.NewRecorder()
+	conf, err := traceconfig.Load("/does/not/exists.yaml")
+	assert.NoError(t, err)
+
+	recv := NewHTTPReceiver(conf, nil, nil, nil)
+	recv.buildMux().ServeHTTP(rec, req)
+	responseBody, err := ioutil.ReadAll(rec.Result().Body)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if string(responseBody) != "OK" {
+		t.Fatalf("proxy failed %s", responseBody)
+	}
+}
+
+func TestTelemetryProxyMultipleEndpoitns(t *testing.T) {
```

Comment:
```suggestion
func TestTelemetryProxyMultipleEndpoints(t *testing.T) {
```
Please check for this typo. You've made it in other places too 🙂